src/EventListener/HeaderReplayStopPropagationListener.php

```diff
@@ -18,40 +18,30 @@
 /**
  * Class HeaderReplayStopPropagationListener.
  *
- * This listener is registered with a very high priority on both, the
- * kernel.response and the kernel.terminate listeners. It makes sure
- * it stops propagation of these events if the current request is a preflight
- * request. It never makes sense to modify a preflight request by one of these
- * events as the bundle provides its own event for this.
+ * This listener is registered with a very high priority on the kernel.terminate
+ * listeners. It makes sure it stops propagation of these events if the current
+ * request is a preflight request. kernel.terminate events are usually listeners
+ * that do "heavy" actions so we certainly do not want these to be executed
+ * during a preflight request.
  * If you really need to do this for whatever reason, register a listener
  * with an even higher priority.
+ *
+ * This is not relevant if you use the Symfony HttpCache because kernel.terminate
+ * will not be called during the preflight request (same process). It is relevant
+ * if you use a separate reverse proxy such as Varnish and spawn two separate
+ * PHP processes. It is enabled by default so it works out of the box.
  */
 class HeaderReplayStopPropagationListener
 {
-    /**
-     * @param FilterResponseEvent $event
-     */
-    public function onKernelResponse(FilterResponseEvent $event)
-    {
-        $this->stopPropagationIfApplicable($event);
-    }
-
     /**
      * @param PostResponseEvent $event
      */
     public function onKernelTerminate(PostResponseEvent $event)
     {
-        $this->stopPropagationIfApplicable($event);
-    }
-
-    /**
-     * @param KernelEvent $event
-     */
-    private function stopPropagationIfApplicable(KernelEvent $event)
-    {
-        $request = $event->getRequest();
-
-        if (in_array(HeaderReplayListener::CONTENT_TYPE, $request->getAcceptableContentTypes(), true)) {
+        if (in_array(
+            HeaderReplayListener::CONTENT_TYPE,
```

Comment:
But Varnish is not sending the `application/vnd.t42.header-replay` content type, is it?

Reply:
It has to.

Reply:
So how do I configure Varnish then?

Reply:
https://github.com/terminal42/header-replay-bundle#varnish